rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterFragmentInteractor.java

```diff
@@ -140,4 +139,26 @@ private EventClient saveEventClient(JSONObject jsonForm, String encounterType, S
 
         return new EventClient(dbEvent, dbClient);
     }
+
+    /**
+     * Check whether to save and return to the patient register of save and proceed to Start RDT capture form
+     *
+     * @param jsonForm The form's JSONObject string representation
+     * @return true if continuing to RDT form else, return false
+     */
+    public boolean continuingToRDTForm(String jsonForm) throws JSONException {
+        JSONObject formJsonObject = new JSONObject(jsonForm);
```

Comment:
Instead of calling this from the fragment's presenter, you could alternatively have it called by the `onPostExecute` of the `SaveFormTask` after checking that encounterType is patient registration, and make this a private method.

Again, this moves all this logic to the interactor, making it more testable

Reply:
It will also be easier to create the Patient object from within the `onPostExecute` and thus launch the rdt form with patient details populated.

Reply:
Or you can use global variables to populate fields needed for creating the Patient object

Reply:
I think the `saveForm()` should remain as it is and have one sole task of saving the form. 
I' can retrieve the entityId set in the `saveEventClient()` and use that and details from the jsonForm to create a _Patient_ object